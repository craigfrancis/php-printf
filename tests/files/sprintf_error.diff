     *** Testing sprintf() : error conditions ***
     
     -- Testing sprintf() function with Zero arguments --
004- sprintf() expects at least 1 argument, 0 given
004+ Too few arguments to function user_sprintf(), 0 passed in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_error.php on line 7 and exactly 1 expected
     
     -- Testing sprintf() function with less than expected no. of arguments --
007- 2 arguments are required, 1 given
008- 3 arguments are required, 2 given
009- 4 arguments are required, 3 given
010- 3 arguments are required, 1 given
011- 4 arguments are required, 2 given
012- 4 arguments are required, 1 given
013- 101 arguments are required, 1 given
014- Missing format specifier at end of string
015- Done
007+ 
008+ Fatal error: Uncaught ValueError: The arguments array must contain 1 items, 0 given in /Volumes/WebServer/Projects/php.fprint/functions.php:15
009+ Stack trace:
010+ #0 /Volumes/WebServer/Projects/php.fprint/functions.php(15): vsprintf('%s', Array)
011+ #1 /Volumes/WebServer/Projects/php.fprint/functions.php(19): user_vsprintf('%s', Array)
012+ #2 /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_error.php(21): user_sprintf('%s')
013+ #3 {main}
014+   thrown in /Volumes/WebServer/Projects/php.fprint/functions.php on line 15
