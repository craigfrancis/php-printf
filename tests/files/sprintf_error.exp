*** Testing sprintf() : error conditions ***

-- Testing sprintf() function with Zero arguments --
sprintf() expects at least 1 argument, 0 given

-- Testing sprintf() function with less than expected no. of arguments --
2 arguments are required, 1 given
3 arguments are required, 2 given
4 arguments are required, 3 given
3 arguments are required, 1 given
4 arguments are required, 2 given
4 arguments are required, 1 given
101 arguments are required, 1 given
Missing format specifier at end of string
Done