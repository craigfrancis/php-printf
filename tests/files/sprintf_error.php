<?php
echo "*** Testing sprintf() : error conditions ***\n";

// Zero arguments
echo "\n-- Testing sprintf() function with Zero arguments --\n";
try {
    var_dump( user_sprintf() );
} catch (TypeError $e) {
    echo $e->getMessage(), "\n";
}

echo "\n-- Testing sprintf() function with less than expected no. of arguments --\n";
$format1 = '%s';
$format2 = '%s%s';
$format3 = '%s%s%s';
$arg1 = 'one';
$arg2 = 'two';

// with one argument less than expected
try {
    var_dump( user_sprintf($format1) );
} catch (\ArgumentCountError $e) {
    echo $e->getMessage(), "\n";
}
try {
    var_dump( user_sprintf($format2,$arg1) );
} catch (\ArgumentCountError $e) {
    echo $e->getMessage(), "\n";
}
try {
    var_dump( user_sprintf($format3,$arg1,$arg2) );
} catch (\ArgumentCountError $e) {
    echo $e->getMessage(), "\n";
}

// with two argument less than expected
try {
    var_dump( user_sprintf($format2) );
} catch (\ArgumentCountError $e) {
    echo $e->getMessage(), "\n";
}
try {
    var_dump( user_sprintf($format3,$arg1) );
} catch (\ArgumentCountError $e) {
    echo $e->getMessage(), "\n";
}

// with three argument less than expected
try {
    var_dump( user_sprintf($format3) );
} catch (\ArgumentCountError $e) {
    echo $e->getMessage(), "\n";
}

try {
    var_dump(user_sprintf('%100$d %d'));
} catch (\ArgumentCountError $e) {
    echo $e->getMessage(), "\n";
}

try {
    var_dump(user_sprintf("foo %", 42));
} catch (ValueError $e) {
    echo $e->getMessage(), "\n";
}

echo "Done";
?>
