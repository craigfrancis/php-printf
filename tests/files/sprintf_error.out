*** Testing sprintf() : error conditions ***

-- Testing sprintf() function with Zero arguments --
Too few arguments to function user_sprintf(), 0 passed in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_error.php on line 7 and exactly 1 expected

-- Testing sprintf() function with less than expected no. of arguments --

Fatal error: Uncaught ValueError: The arguments array must contain 1 items, 0 given in /Volumes/WebServer/Projects/php.fprint/functions.php:15
Stack trace:
#0 /Volumes/WebServer/Projects/php.fprint/functions.php(15): vsprintf('%s', Array)
#1 /Volumes/WebServer/Projects/php.fprint/functions.php(19): user_vsprintf('%s', Array)
#2 /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_error.php(21): user_sprintf('%s')
#3 {main}
  thrown in /Volumes/WebServer/Projects/php.fprint/functions.php on line 15